lib/puppet/provider/sensu_api.rb

```diff
@@ -260,7 +259,10 @@ def self.version_cmp(v)
       @current_version = version
     end
     return true if @current_version.nil?
-    return Gem::Version.new(@current_version) >= Gem::Version.new(v)
+    Gem::Version.new(@current_version) >= Gem::Version.new(v)
+  rescue ArgumentError => e
+    Puppet.debug "Unable to compare version #{@current_version} with needed version #{v}: #{e}"
```

Comment:
Document this behavior along with an explanation and link to the issue

Reply:
Comments added.